lib/output.js

```diff
@@ -1185,6 +1186,35 @@ function OutputStream(options) {
         output.print(self.getValue());
     });
     DEFPRINT(AST_String, function(self, output){
+        var parent = output.parent(1);
+        // Check if there is a need to break directive by inserting an additional semicolon
+        if (parent instanceof AST_Scope) {
+            var print_semicolon = false;
+            // Check if string is the first in body or after directive
+            for (var i = 0; i < parent.body.length; i++) {
+                if (parent.body[i] instanceof AST_SimpleStatement && parent.body[i].body instanceof AST_String) {
+                    print_semicolon = parent.body[i].body === self;
+                    break;
+                }
+                if (!(parent.body[i] instanceof AST_Directive)) {
+                    break; // Breaking directive prologue
+                }
+            }
+            if (print_semicolon) {
+                var str = output.encode_string(self.getValue(), self.quote);
+                // Only output semicolons if string might be a valid directive
+                if (str.indexOf("\\") === -1) {
+                    var last = output.last();
+                    // Extra semicolon might be needed to break directive prologue
+                    if (last !== null && /[;{]\s*$/.test(last)) {
+                        output.force_semicolon();
+                    }
+                    output.force_semicolon();
+                }
+                output.print(str);
+                return;
+            }
+        }
```

Comment:
Trees could be altered (though compression typically will remove none or all). And ultimately it will only move logic (but some places might be better as current check is done on what might be considered a loop).